### PR TITLE
docs: add composable data ecosystem concept

### LINKED ISSUE #6618: docs: conceptual explanation of Ibis, ADBC, Substrait relationships
today I saw that one can use Polars to query Snowflake directly using ADBC https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.read_database.html

and started thinking what should be the relationship between ADBC and Ibis. I saw various closed PRs on the Ibis side (https://github.com/ibis-project/ibis/pull/5010, https://github.com/ibis-project/ibis/pull/4267) so initially this was not clear to me.
 
Digging a bit more, I also came across Substrait and initially it was not obvious to me what the relationship between Substrait and ADBC was. I finally found this:
 
> ADBC allows applications to pass queries either as SQL strings or Substrait plans.
 
https://substrait.io/community/powered_by/
 
Do you have any more insight on what are the overall plans and roadmap in this space of the whole Ibis -> Backend -> Substrait -> ADBC -> Database tower?

_Originally posted by @astrojuanlu in https://github.com/ibis-project/ibis/discussions/6617_

### PR DESCRIPTION
**Pull request by lostmygithubaccount**
## Description of changes

add conceptual article for the composable data ecosystem, explaining the relationship between Ibis and other key projects (Apache Arrow, ADBC, Substrait)

very WIP. some TODOs:

- [x] finish prose
- [x] formatting
- [x] share w/ other projects for review/feedback/contribution

I plan to finish my first pass on this ~tomorrow and share w/ other projects. some open questions:

- is it worth mentioning Flight and FlightSQL? I kinda feel like it's confusing and focusing on ADBC as the connectivity layer is better
- are there other projects we should mention? perhaps SQLGlot fits in here?
- Substrait: working through the finer points of why SQL isn't sufficient and Substrait is needed

## Issues closed

closes #6618

### CODE REVIEW

**Comment by lostmygithubaccount** [comment]
first pass done, need to re-read and share w/ other project maintainers etc.

**Review by cpcloud — APPROVED** [review]
LGTM, thank you!

File: docs/concepts/composable-ecosystem.qmd [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+for data interchange between with other Python dataframe libraries and
+visualization libraries. This is largely efficient because most libraries
+support Apache Arrow as their in-memory data format, making interchange between
+them cheap and efficient.
+
+## Apache Arrow Database Connectivity (ADBC)
+
+[Apache Arrow Database Connectivity
+(ADBC)](https://arrow.apache.org/docs/format/ADBC.html) is a relatively new
+standard for database connectivity. It is a low-level protocol for exchanging
```
[the remark this answers is not in the document]
Comment (lidavidm) [reply docs/concepts/composable-ecosystem.qmd:74]: ah whoops, I didn't get to this in time

[the remark this answers is not in the document]
Comment (lostmygithubaccount) [reply docs/concepts/composable-ecosystem.qmd:74]: I'll make a PR to update

[the remark this answers is not in the document]
Comment (lostmygithubaccount) [reply docs/concepts/composable-ecosystem.qmd:74]: PR here: https://github.com/ibis-project/ibis/pull/7941

**Pull request merged**